doc/articles/howto-consume-webservices.md

```diff
@@ -0,0 +1,1849 @@
+
+    namespace TheCatApiClient.Shared.WebServices
+    {
+        public class FavoritesApi : WebApiBase
+        {
+            private Dictionary<string, string> _defaultHeaders = new Dictionary<string, string> {
+                {"accept", "application/json" },
+                {"x-api-key", "{YOUR-API-KEY}"}
+            };
+
+            // Insert GetAll and Get below here
+
+            // Insert Add below here
+
+            // Insert Delete below here
+            }
+        }
+    }
+    ```
+
+1. To add support for retrieving all of your favorites, or a specific favorite, locate the comment **// Insert GetAll and Get below here** and replace it with the following code:
+
+    ```csharp
+    // Insert GetAll and Get below here
+    public async Task<IEnumerable<Favorite>> GetAll()
+    {
+        var result = await this.GetAsync(
+            $"https://api.thecatapi.com/v1/favourites",
+            _defaultHeaders);
+
+        if (result != null)
+        {
+            return JsonSerializer.Deserialize<IEnumerable<Favorite>>(result);
+        }
+
+        return new List<Favorite>();
+    }
+
+    public async Task<Favorite> Get(int id)
+    {
+        var result = await this.GetAsync(
+            $"https://api.thecatapi.com/v1/favourites/{id}",
+            _defaultHeaders);
+
+        if (result != null)
+        {
+            return JsonSerializer.Deserialize<Favorite>(result);
+        }
+
+        return null;
+    }
+    ```
+
+    These methods are very similar to the methods you have implemented earlier - **GetAll** returns a collection of the users **Favorite** instances, whereas **Get** returns a single **Favorite** by its ID.
+
+    > [!TIP]
+    > You can review the favorites service documentation here - [/favorites](https://docs.thecatapi.com/api-reference/favourites/)
+
+1. To add support for adding an image to your favorites, locate the comment **// Insert Add below here** and replace it with the following code:
+
+    ```csharp
+    // Insert Add below here
+    public async Task<Response> Add(CatImage image)
+    {
+        var result = await this.PostAsync(
+            $"https://api.thecatapi.com/v1/favourites",
+            JsonSerializer.Serialize(
+                new Dictionary<string, string>
+                {
+                    { "image_id", image.Id },
+                    { "sub_id", "uno-client" }
+                }),
+            _defaultHeaders);
+
+        if (result != null)
+        {
+            return JsonSerializer.Deserialize<Response>(result);
+        }
+
+        return null;
+    }
+    ```
+
+    In order to add an image to your favorites, the api expects a JSON payload that contains the image ID and an optional **sub_id** value. The **Add** methods constructs this using a **Dictionary** that is then serialized to JSON and submitted as the payload. Rather than returning a **Favorite**, it returns a **Response** instance which contains the new Favorite ID.
```

Comment:
```suggestion
    In order to add an image to your favorites, the API expects a JSON payload that contains the image ID and an optional **sub_id** value. The **Add** methods constructs this using a **Dictionary** that is then serialized to JSON and submitted as the payload. Rather than returning a **Favorite**, it returns a **Response** instance which contains the new Favorite ID.
```